homeassistant/components/accuweather/sensor.py

```diff
@@ -1,83 +1,104 @@
 """Support for the AccuWeather service."""
+from __future__ import annotations
+
+from typing import Any, cast
+
 from homeassistant.components.sensor import SensorEntity
-from homeassistant.const import (
-    ATTR_ATTRIBUTION,
-    ATTR_DEVICE_CLASS,
-    CONF_NAME,
-    DEVICE_CLASS_TEMPERATURE,
-)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ATTRIBUTION, CONF_NAME, DEVICE_CLASS_TEMPERATURE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
+from . import AccuWeatherDataUpdateCoordinator
 from .const import (
     ATTR_FORECAST,
-    ATTR_ICON,
-    ATTR_LABEL,
     ATTRIBUTION,
     COORDINATOR,
     DOMAIN,
-    FORECAST_DAYS,
     FORECAST_SENSOR_TYPES,
     MANUFACTURER,
+    MAX_FORECAST_DAYS,
     NAME,
-    OPTIONAL_SENSORS,
     SENSOR_TYPES,
 )
 
 PARALLEL_UPDATES = 1
 
 
-async def async_setup_entry(hass, config_entry, async_add_entities):
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
     """Add AccuWeather entities from a config_entry."""
-    name = config_entry.data[CONF_NAME]
+    name = entry.data[CONF_NAME]
 
-    coordinator = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
+    coordinator = hass.data[DOMAIN][entry.entry_id][COORDINATOR]
 
     sensors = []
     for sensor in SENSOR_TYPES:
         sensors.append(AccuWeatherSensor(name, sensor, coordinator))
 
     if coordinator.forecast:
         for sensor in FORECAST_SENSOR_TYPES:
-            for day in FORECAST_DAYS:
+            for day in range(MAX_FORECAST_DAYS + 1):
                 # Some air quality/allergy sensors are only available for certain
                 # locations.
                 if sensor in coordinator.data[ATTR_FORECAST][0]:
                     sensors.append(
                         AccuWeatherSensor(name, sensor, coordinator, forecast_day=day)
                     )
 
-    async_add_entities(sensors, False)
+    async_add_entities(sensors)
 
 
 class AccuWeatherSensor(CoordinatorEntity, SensorEntity):
     """Define an AccuWeather entity."""
 
-    def __init__(self, name, kind, coordinator, forecast_day=None):
+    coordinator: AccuWeatherDataUpdateCoordinator
+
+    def __init__(
+        self,
+        name: str,
+        kind: str,
+        coordinator: AccuWeatherDataUpdateCoordinator,
+        forecast_day: int | None = None,
+    ) -> None:
         """Initialize."""
         super().__init__(coordinator)
+        if forecast_day is None:
+            self._description = SENSOR_TYPES[kind]
+            if kind == "Precipitation":
+                self._sensor_data = coordinator.data["PrecipitationSummary"][kind]
```

Comment:
```suggestion
                self._sensor_data: dict[str, Any] = coordinator.data["PrecipitationSummary"][kind]
```
Or outside `if` just add type definition:
```python
self._sensor_data: dict[str, Any]
```